server/src/main/java/org/elasticsearch/cluster/routing/OperationRouting.java

```diff
@@ -127,7 +129,9 @@ public GroupShardsIterator<ShardIterator> searchShards(
                 set.add(PlainShardIterator.allSearchableShards(iterator));
             }
         }
-        return GroupShardsIterator.sortAndCreate(new ArrayList<>(set));
+        var res = new ArrayList<>(set);
+        CollectionUtil.timSort(res);
```

Comment:
@javanna see my other comment this one in particular seems unnecesssary.  We already sort the iterators in the search action, we shouldn't need to sort them here as well?